packages/workflow/src/internals.ts

```diff
@@ -577,6 +686,34 @@ export class Activator implements ActivationHandler {
     }
   }
 
+  public dispatchBufferedUpdates(): void {
+    const bufferedUpdates = this.bufferedUpdates;
+    while (bufferedUpdates.length) {
+      const foundIndex = bufferedUpdates.findIndex((update) => update.name && this.updateHandlers.has(update.name));
+      if (foundIndex === -1) {
+        // No buffered Updates have a handler yet.
+        break;
+      }
+      const [update] = bufferedUpdates.splice(foundIndex, 1);
+      this.doUpdate(update);
+    }
+  }
+
+  public rejectBufferedUpdates(): void {
+    while (this.bufferedUpdates.length) {
+      const update = this.bufferedUpdates.shift();
+      if (update) {
+        if (!update.id) {
+          throw new TypeError('Missing activation update id');
+        }
+        this.rejectUpdate(
+          update.id,
+          ApplicationFailure.nonRetryable(`No registered handler for update: ${update.name}`)
+        );
+      }
+    }
```

Comment:
If you want to follow the pattern we have in the other methods, you could throw a type error if the id or name are not truthy.